systems/plants/test/testSurfaceTangentsMex.m

```diff
@@ -0,0 +1,32 @@
+function testSurfaceTangentsMex
+
+checkDependency('bullet');
+
+%build an atlas model 
+p = RigidBodyManipulator(fullfile('../../../examples/Atlas/urdf/atlas_minimal_contact.urdf'));
+if(p.mex_model_ptr == 0)
+    disp('testSurfaceTangentsMex: no mex model pointer... nothing to test');
+    return;
+end
+
+%random initial pose
+nq = p.getNumPositions();
+q0 = randn(nq, 1); 
```

Comment:
Better to start using getRandomConfiguration(). This handles quat floating joints better (quat should have norm 1). Also it will give you a uniformly distributed rotation for floating joints.